tests/cnf/core/network/sriov/tests/mlxSecureBoot.go

```diff
@@ -0,0 +1,195 @@
+			Expect(err).ToNot(HaveOccurred(), "Failed to validate if SecureBoot is enabled")
+
+			if isSecureBootEnabled {
+				configureSecureBoot(bmcClient, "disable")
+			}
+
+			By("Disabling Mellanox firmware and wait for the cluster becomes stable")
+			err = sriovenv.ConfigureSriovMlnxFirmwareOnWorkersAndWaitMCP(workerNodeList, sriovInterfacesUnderTest[0], false, 0)
+			Expect(err).ToNot(HaveOccurred(), "Failed to configure Mellanox firmware")
+		})
+
+		It("End-to-End SR-IOV Configuration and Validation", reportxml.ID("77014"), func() {
+			By("Creating SR-IOV policy")
+			const sriovAndResourceNameSecureBoot = "securebootpolicy"
+
+			sriovPolicy := sriov.NewPolicyBuilder(
+				APIClient,
+				sriovAndResourceNameSecureBoot,
+				NetConfig.SriovOperatorNamespace,
+				sriovAndResourceNameSecureBoot,
+				2,
+				sriovInterfacesUnderTest[:1],
+				map[string]string{"kubernetes.io/hostname": workerNodeList[0].Definition.Name})
+
+			err := sriovenv.CreateSriovPolicyAndWaitUntilItsApplied(sriovPolicy, tsparams.MCOWaitTimeout)
+			Expect(err).ToNot(HaveOccurred(), "Failed to configure SR-IOV policy")
+
+			By("Creating SR-IOV network")
+
+			_, err = sriov.NewNetworkBuilder(
+				APIClient, sriovAndResourceNameSecureBoot, NetConfig.SriovOperatorNamespace,
+				tsparams.TestNamespaceName, sriovAndResourceNameSecureBoot).WithStaticIpam().
+				WithIPAddressSupport().WithLogLevel(netparam.LogLevelDebug).Create()
+			Expect(err).ToNot(HaveOccurred(), "Failed to create SR-IOV network")
+
+			By("Creating test pods and checking connectivity")
+			err = sriovenv.CreatePodsAndRunTraffic(workerNodeList[0].Definition.Name, workerNodeList[0].Definition.Name,
+				sriovAndResourceNameSecureBoot, sriovAndResourceNameSecureBoot, "", "",
+				[]string{tsparams.ClientIPv4IPAddress}, []string{tsparams.ServerIPv4IPAddress})
+			Expect(err).ToNot(HaveOccurred(), "Failed to test connectivity between test pods")
+
+			By("Removing SR-IOV node policy")
+			err = sriovPolicy.Delete()
+			Expect(err).ToNot(HaveOccurred(), "Failed to delete SR-IOV policy")
+
+			err = netenv.WaitForSriovAndMCPStable(
+				APIClient, tsparams.MCOWaitTimeout, tsparams.DefaultStableDuration,
+				NetConfig.CnfMcpLabel, NetConfig.SriovOperatorNamespace)
+			Expect(err).ToNot(HaveOccurred(), "Failed to wait for MCP and SR-IOV update")
+
+			By("Validation that totalvfs is still the same after the policy removal")
+			sriovNodeState := sriov.NewNetworkNodeStateBuilder(
+				APIClient, workerNodeList[0].Object.Name, NetConfig.SriovOperatorNamespace)
+			err = sriovNodeState.Discover()
+			Expect(err).ToNot(HaveOccurred(), "Failed to discover SR-IOV node state")
+
+			currentTotalVFs, err := sriovNodeState.GetTotalVFs(sriovInterfacesUnderTest[0])
+			Expect(err).ToNot(HaveOccurred(), "Failed to get totalvfs on SR-IOV node")
+
+			Expect(currentTotalVFs).To(BeNumerically("==", totalVFs))
+		})
+	})
+
+func configureSecureBoot(bmcClient *bmc.BMC, action string) {
```

Comment:
Should these two functions be moved out of the test file?

Reply:
moved